homeassistant/components/soundtouch/media_player.py

```diff
@@ -78,11 +83,16 @@
     | SUPPORT_PLAY_MEDIA
 )
 
+SOURCE_SCHEMA = vol.Schema(
+    {vol.Required(CONF_SOURCE): cv.string, vol.Required(CONF_NAME): cv.string}
+)
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_HOST): cv.string,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_RESOURCES): vol.All(cv.ensure_list, [SOURCE_SCHEMA]),
```

Comment:
Why do we need this option? If we know what the device supports, we can just return that list of sources in the media player entity source list.

Reply:
https://github.com/CharlesBlonde/libsoundtouch ,    this library is not updated 2 years.   no api to get sources that device supported.
so config sources to configuration.yaml

Reply:
would it make sense to bump libsoundtouch to 0.8.0 since HA is currently using an even older version?

Reply:
btw - this might be an alternative in future? Just found it https://github.com/trunet/aiobosest

Reply:
We like asyncio! If that works it might be a very nice replacement! The repo looks a bit dormant though.

Reply:
> would it make sense to bump libsoundtouch to 0.8.0 since HA is currently using an even older version?

in this pr, libsoundtouch lib already be updated to 0.8.0

Reply:
Since the sources could be fetched from the api if the library would support it, I don't think adding this config option is a good step forward. We want to avoid config options as much as possible.

Please update or change library first, in a separate PR, and then we see if this PR is still needed.

Reply:
there is a pending PR over at libsoundtouch which would add a source list, but the maintainer didn't respond to PRs in over a year. https://github.com/CharlesBlonde/libsoundtouch/pull/34

> The repo looks a bit dormant though.

Yes, the author of it commented that he wrote it for HA back then but the guys implementing the HA component went with libsoundtouch instead, which is why he stopped working on it. But he seems to be interested in working on it and implement the missing features if we (HA) considers to switch to it. 
https://github.com/CharlesBlonde/libsoundtouch/issues/38

I'm not saying we have to or should, especially since libsoundtouch is well documented and it's in always a pitty if well used projects either have to be forked or abandoned.

Reply:
If the maintainer of https://github.com/trunet/aiobosest is active and the maintainer of https://github.com/CharlesBlonde/lib soundtouch is not, I suggest we change to the active library.